hack/boilerplate/boilerplate.py

```diff
@@ -70,25 +102,52 @@ def file_passes(filename, extension, ref, regexs):
 
     return True
 
-def main():
-    if len(sys.argv) < 3:
-        PrintError("usage: %s extension FILENAME [FILENAMES]" % sys.argv[0])
-        return False
-
-    basedir = os.path.dirname(os.path.abspath(__file__))
-
-    extension = sys.argv[1]
-    # argv[0] is the binary, argv[1] is the extension (go, sh, py, whatever)
-    filenames = sys.argv[2:]
-
-    ref_filename = basedir + "/boilerplate." + extension + ".txt"
-    try:
-        ref_file = open(ref_filename, 'r')
-    except:
-        # No boilerplate template is success
-        return True
-    ref = ref_file.read().splitlines()
-
+def valid_file_extension(filename, extensions):
+    for ext in extensions:
+        if filename.endswith("." + ext):
+            return True
+    return False
+
+def normalize_files(files):
```

Comment:
Maybe I don't entirely understand how `os.walk` works, but it seems you sufficiently guard against this in `get_files`. What am I missing?